System/DataModule.h

```diff
@@ -333,6 +360,16 @@ namespace RTE {
 #pragma endregion
 
 #pragma region Entity Mapping
+		/// <summary>
+		/// Shared method for filling a list with all previously read in (defined) Entities which are or are not associated with a specific group or groups.
+		/// </summary>
+		/// <param name="entityList">Reference to a list which will get all matching Entities added to it. Ownership of the list or the Entities placed in it are NOT transferred!</param>
+		/// <param name="type">The name of the least common denominator type of the Entities you want. "All" will look at all types.</param>
+		/// <param name="groups">The group or groups to look for.</param>
+		/// <param name="excludeGroups">Whether Entities belonging to the specified group or groups should be excluded.</param>
+		/// <returns>Whether any Entities were found and added to the list.</returns>
+		bool GetAllOfOrNotOfGroups(std::list<Entity *> &entityList, const std::string &type, const std::variant<const std::string, const std::vector<std::string>> &groups, bool excludeGroups);
```

Comment:
The variant thing is neat, but I don't think it's really better than just making a vector in the GetAllOfGroup and GetAllNotOfGroup versions. Especially since you make that vector in this method anyway.

Also, why is the string part of the variant not a reference?